packages/firestore/src/local/shared_client_state.ts

```diff
@@ -732,6 +896,35 @@ export class WebStorageSharedClientState implements SharedClientState {
       mutationBatch.error
     );
   }
+
+  private handleQueryTargetEvent(
+    targetMetadata: QueryTargetMetadata
+  ): Promise<void> {
+    return this.syncEngine.applyTargetState(
+      targetMetadata.targetId,
+      targetMetadata.state,
+      targetMetadata.error
+    );
+  }
+
+  private handleClientStateEvent(
+    clientState: RemoteClientState
+  ): Promise<void> {
+    const existingTargets = this.activeClients[clientState.clientId]
+      ? this.activeClients[clientState.clientId].activeTargetIds
+      : targetIdSet();
+    this.activeClients[clientState.clientId] = clientState;
+
+    let newTargets = Promise.resolve();
+    clientState.activeTargetIds.forEach(async targetId => {
+      if (!existingTargets.has(targetId)) {
+        newTargets = newTargets.then(() =>
+          this.syncEngine.applyTargetState(targetId, 'pending')
```

Comment:
I am expecting that only the master tab cares about new targets, and it should also care about removed targets (so it knows to stop listening).  So I am thinking that this shouldn't use applyTargetState() and instead we should have new method(s) to notify the master of new/removed targets (though we need to somehow aggregate across all client states to know if it's new or not).

Reply:
I added a new method to deal with this (appleActiveTargetsChange as suggested by our Kirkland team). I am now also `de-duping` these notifications in SharedClientState.